legal-api/src/legal_api/services/authz.py

```diff
@@ -324,23 +324,26 @@ def has_roles(jwt: JwtManager, roles: List[str]) -> bool:
 
 
 # pylint: disable=(too-many-arguments,too-many-locals
-def is_allowed(state: Business.State,
+def is_allowed(business: Business,
+               state: Business.State,
                filing_type: str,
                legal_type: str,
                jwt: JwtManager,
-               sub_filing_type: str = None):
+               sub_filing_type: str = None,
+               filing_id: int = None):
     """Is allowed to do filing."""
-    user_role = 'general'
-    if jwt.contains_role([STAFF_ROLE, SYSTEM_ROLE, COLIN_SVC_ROLE]):
-        user_role = 'staff'
-
-    allowable_filing = ALLOWABLE_FILINGS.get(user_role, {}).get(state, {}).get(filing_type, {})
-    allowable_filing_legal_types = allowable_filing.get('legalTypes', [])
+    allowable_filings = get_allowed_filings(business, state, legal_type, jwt)
```

Comment:
`get_allowed_filings` needs to be updated such that it is possible to turn off the logic that filters out allowed filings that already have a filing in `DRAFT` status.  Perhaps via an additional parameter in the `get_allowed_filings` function.  

Once that is in place, in this function(`is_allowed`), maybe you can pull back the filing(`filing_id`) if there is one and check if it's in DRAFT state.  If it's in DRAFT state and is a filing returned by `allowable_filings`, return True.